pallets/task/src/lib.rs

```diff
@@ -166,14 +170,20 @@ pub mod pallet {
 		/// Event for creation of task [AccountID, hash id]
 		TaskCreated(T::AccountId, T::Hash),
 
+		/// Event for updating existing task [AccountID, hash id]
+		TaskUpdated(T::AccountId, T::Hash),
+
 		/// Task assigned to new account [AccountID, hash id]
 		TaskAssigned(T::AccountId, T::Hash),
 
 		/// Task completed by assigned account [AccountID, hash id]
 		TaskCompleted(T::AccountId, T::Hash),
```

Comment:
trying to understand this, A task when marked completed emits event TaskCompleted so that frontend gets notified but the taskStatus will be still InProgress, so if someone checks status of task via RPC  Storage query it want see that it is marked completed.  Can't this lead to inconsistency for UI? Specially when they just don't rely on events only.

Reply:
I updated the design. 

We don't rely on UI events in the front-end, but it's good to be consistent.